doc/changelog/11-standard-library/19310-stdlib-logical-name.rst

```diff
@@ -0,0 +1,10 @@
+- **Changed:**
+  the requirement prefix of the standard library from ``Coq`` to
+  ``Stdlib``. For backward compatibility with older versions of Coq
+  ``Coq`` is immediatly translated to ``Stdlib`` with a warning. When
+  droping support for Coq <= 8.20, a simple script such as ``for f in
+  $(find . -name "*.v") ; do sed -i $f -e 's/From Coq Require/From
+  Stdlib Require/;s/Coq[.]\([a-zA-Z]\)/Stdlib[.]\1/g' ; done`` should
+  be enough to adapt your code
```

Comment:
Should we remind users of the usual support for silencing the warning project-wise in `_CoqProject` before dropping support for Coq <= 8.20?

BTW, it makes me think that it could be worthwhile to create a how-to guide on migrating Coq libraries to newer Coq versions. I feel like lots of folklore knowledge has been accumulated about this.

Reply:
> Should we remind users of the usual support for silencing the warning project-wise in `_CoqProject` before dropping support for Coq <= 8.20?

Why not, done

> BTW, it makes me think that it could be worthwhile to create a how-to guide on migrating Coq libraries to newer Coq versions. I feel like lots of folklore knowledge has been accumulated about this.

I agree. I don't have time to write this myself but I'd be happy to review it if anyone comes with something.